internal/util/credentials.go

```diff
@@ -145,11 +145,31 @@ func ParseAndSetSecretMapFromMigSecret(secretmap map[string]string) (map[string]
 // of a migration volume request. The migration secret carry a field
 // called `key` which is the equivalent of `userKey` which is what we
 // check here for identifying the secret.
-func IsMigrationSecret(passedSecretMap map[string]string) bool {
+func isMigrationSecret(passedSecretMap map[string]string) bool {
 	// the below 'nil' check is an extra measure as the request validators like
 	// ValidateNodeStageVolumeRequest() already does the nil check, however considering
 	// this function can be called independently with a map of secret values
 	// it is good to have this check in place, also it gives clear error about this
 	// was hit on migration request compared to general one.
 	return len(passedSecretMap) != 0 && passedSecretMap[migUserKey] != ""
 }
+
+// NewUserCredentialsWithMigration takes secret map from the request and validate it is
+// a migration secret, if yes, it continues to create CR from it after parsing the migration
+// secret. If it is not a migration it will continue the attempt to create credentials from it
+// without parsing the secret. This function returns credentials and error.
+func NewUserCredentialsWithMigration(passedSecretMap map[string]string) (*Credentials, error) {
```

Comment:
that is `passedSecretMap`? Do you mean 'passed' as in the argument to the function? That seems rather redundant, and just `secrets` works fine everywhere else too.

Reply:
done.